lib/assessment.js

```diff
@@ -8,32 +8,73 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
+var min = function min(previousValue, currentValue, index, array) {
+    var result = previousValue;
+    if(currentValue < result) result = currentValue;
+    return result;
+
 
 };
 
+var array = [5, 3, 2, 4]
+console.log(array.reduce(min))
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
+var sum = function sum(previousValue, currentValue, index, array) {
+       var result = previousValue;
+     result += currentValue;
+    return result;
+
 
 };
 
+var array = [5, 74, 2, 4]
+console.log(array.reduce(sum));
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
 
+ var result = true;
+
+ array.forEach(function(currentValue, index, array){
+     result = result && callback(currentValue, index, array);
+ });
+ return result;
+
+};
```

Comment:
This is a great answer and shows a command of JavaScript. However, `any`'s logic is incorrect.

The question asks for a function that returns `true` if the callback returns `true` for **any** element in the array. This suggests that we should be using a logical OR (`||`), not a logical AND (`&&`). A logical AND would be suitable for determining whether a callback returns `true` for **all** elements of the array. 

To know what our initial value of `result` should be, we consult the [truth table for the OR operation](https://en.wikipedia.org/wiki/Logical_disjunction#Truth_table). As we see on row 2 of the truth table, `true || false` evaluates to `true`. Therefore, if we initialize `result` to `true`, `any` will return `true` regardless of what the callback returns for each element of the list -- this is not what we want. There's only one more value to consider: `false`.

We see on lines 4 and 3 of the truth table that `false || false` yields `false`, and `false || true` yields `true`. So, if `result` starts out as `false`, it will remain `false` until it is ORed with `true`. This sounds like what we want. Also, once `result` stores `true`, it will remain `true` no matter what we OR it with. This also sounds like what we want here.

So line 42 should read: `var result = false;` and line 45 should read: `result = result || callback(...);`

3/4 credit.